### ISSUE TITLE
Secondary modal on page not firing the the "ready" and "complete" functions

### ISSUE THREAD

**Issue by carnageX** [Dogfalo/materialize#3723]
So I have 2 modals on a page.  One is activated by a button, which seems to work fine (this is the "Add" modal).  The other is activated by an AngularJS "ng-click" on a FontAwesome <i> tag (this is the "Edit" modal).  

The issue I seem to be having is, that the "Edit" modal doesn't seem to be firing the ready/complete events defined in the leadModal function.  From my ng-click, I call $('#editItemModal').openModal(); to open the modal - which works fine.  However, it doesn't seem to hook into the leanModal function, which I have defined here: 

`$(document).ready(function () {
            $('.modal-trigger').leanModal({
                ready: function () { $('select').material_select(); console.log('modal open'); },
                complete: function () { $('select').material_select('destroy'); console.log('modal close');  }
            });
        });`

When I open/close the Add modal, I get the "modal open" and "modal close" in console.  No such output with the Edit modal. 

I need it to hook in because I'm having a secondary issue with the Material Select input not properly selecting it's item (I think the issue is caused by this, anyway), because I'm creating/setting the contents dynamically.  I'm setting the selected item in AngularJS as such for the edit modal's model: 

`$scope.editItemLoad = function (item) {
            $scope.editItem = item;
            $('#editItemModal').openModal();
            console.log($scope.selectedType);
            $scope.selectedType = item.type;
            console.log($scope.selectedType);
        }`

If I set the "browser-default" class on the Select input, then it works fine and displays my selected item.  
![image](https://cloud.githubusercontent.com/assets/7111043/19062875/5117978e-89c3-11e6-9878-79514e694a57.png)  
![image](https://cloud.githubusercontent.com/assets/7111043/19062888/67094d80-89c3-11e6-842e-091e4eea2dc9.png)

I've also tried manually calling the $('select').material_select(); in my AngularJS editItemLoad function...but it doesn't work (also tried calling $apply but it just blows up and says it's already in progress).  

Any help is appreciated, thanks!  Let me know if there's any more info/code I need to post and I can.

**Comment by trevorrjohn** [comment]
any luck with this? I am experiencing a similar issue.

**Comment by carnageX** [comment]
Unfortunately not for the .  I gave up on the 2 modals, and combined the 2 into 1, and pass in a flag on my ng-click events for if it's an add or an edit (not the greatest solution, but I couldn't get anything to work).  I then use Angular-Materialize to better hook up to the Materialize stuff through Angular - seems to work well.  I didn't use it for the modals...but I did use it for the dropdown, as well as the tooltips and it fixed both issues I was previously having with those.
